addons/l10n_tr_nilvera_einvoice/models/res_partner_category.py

```diff
@@ -0,0 +1,54 @@
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+l10n_tr_official_code_categories = [
+    "res_partner_category_hizmetno",
+    "res_partner_category_mersisno",
+    "res_partner_category_tesisatno",
+    "res_partner_category_telefonno",
+    "res_partner_category_distributorno",
+    "res_partner_category_ticaretsicilno",
+    "res_partner_category_tapdkno",
+    "res_partner_category_bayino",
+    "res_partner_category_aboneno",
+    "res_partner_category_sayacno",
+    "res_partner_category_epdkno",
+    "res_partner_category_subeno",
+    "res_partner_category_pasaportno",
+    "res_partner_category_ureticino",
+    "res_partner_category_ciftcino",
+    "res_partner_category_imalatcino",
+    "res_partner_category_dosyano",
+    "res_partner_category_hastano",
+    "res_partner_category_musterino",
+    "res_partner_category_aracikurumvkn",
+    "res_partner_category_aracikurumetiket",
+]
+
+l10n_tr_official_mandatory_code_categories = [
+    "res_partner_category_mersisno",
+    "res_partner_category_ticaretsicilno",
+]
+
+
+class PartnerCategory(models.Model):
+    _inherit = "res.partner.category"
+
+    def _get_categories_from_xml_ids(self, xml_ids_list):
+        categories = self.env["res.partner.category"]
+        for xml_id in xml_ids_list:
+            categories |= self.env.ref(f"l10n_tr_nilvera_einvoice.{xml_id}")
```

Comment:
small doubt if this will not crash if the xml_id are not present on the db since it may need an update of the module :thinking: